### DISCUSSION TITLE
Failed to restore from backup: All subscriptions and replication slots must be dropped before a backup can be restored.

### DISCUSSION THREAD

**Discussion by sudayavarman** [supabase/supabase#21830] [Troubleshooting]
As the error suggests, you must first drop any current subscriptions or replication slots to restore backups.

You can check those with:

```
SELECT * FROM pg_replication_slots;

SELECT * FROM pg_subscription;
```

You can drop them with:

```
DROP SUBSCRIPTION <subscription>;

SELECT pg_drop_replication_slot(slot_name);
```

NOTE: These are destructive actions. This is fine since you will overwrite your database with a backup.

**Comment by elithubb** [comment]
ERROR: 42501: must be a superuser to terminate superuser process

**Reply by CrewXI** [reply]
same error here, Must be a superuser to terminate superuser processes

**Reply by filipecabaco** [reply]
Could you check which version of CLI you're using?

You can check it with supabase --version

**Comment by elithubb** [comment]
Ongoing issue I am experiencing with reverting backups in my Supabase project.

Despite my efforts, I am still unable to revert to older backups. I would like to provide you with the version of the CLI I am using for your reference:

SQL code:
SELECT version();
Result:
PostgreSQL Version: 15.6
Architecture: aarch64-unknown-linux-gnu
Compiler: GCC 13.2.0, 64-bit

Interestingly, backups are functioning correctly on a simpler project, but I am encountering unresolved issues when attempting to revert backups on a more complex project.  


This is my conversation with Supabase customer service: 
 Subject: Follow-Up on Backup Restoration Issue

Dear Supabase Support Team,

Thank you for your prompt response regarding my issue with restoring the database from a backup. I appreciate your suggestion to disable realtime publications via the publication dashboard.

However, I am still facing challenges due to permission restrictions. When I attempted to terminate the active processes for the replication slots and drop the subscriptions, I encountered the following error:

ERROR: 42501: must be a superuser to terminate superuser process

As a result, I am unable to proceed with removing the replication slots and subscriptions necessary for restoring my database.

Could you please assist me in removing the following replication slots and any related subscriptions so that I can successfully restore my database?

realtime_messages_replication_slot_v2_34_1
supabase_realtime_replication_slot_v2_34_1
Your assistance in this matter would be greatly appreciated.

**Reply by filipecabaco** [reply]
@Ajaxy12 from the description you are providing the postgres version. From your terminal run `supabase --version` because the latest version should have this fixed since this was a direct impact from some realtime changes but both CLI and Realtime have this fixed in the latest version.

PRs for reference:
https://github.com/supabase/cli/pull/3063
https://github.com/supabase/realtime/pull/1279

**Comment by leonaburime-ucla** [comment]
For anyone struggling with this you can restart your project 

Project Settings(left sidebar) -> Restart project

and then quickly revert to a backup

**Comment by CrewXI** [comment]
My project was able to restore without having to restart, not sure what was
done on the backend to make it work.

However I do have another issue where I'm unable to "Restore to a new
project".  what does it mean "restore to new project requires physical
backups"?  if i have scheduled backups to select then physical backups are
already running right?

[image: image.png]

My main goal is to create a database template project that i can quickly
bring up the same table/functions/triggers ect in a new database from the
original template database.

Let me know if you can help me get this Restore to new project working, or
an alternate solution.

Thanks,
